### ISSUE TITLE
Documentation of `Flow.futureFlow`: example needs updating

### ISSUE THREAD

**Issue by akauppi** [akka/akka#31284]
The page [Flow.futureFlow](https://doc.akka.io/docs/akka/current/stream/operators/Flow/futureFlow.html#flow-futureflow) (Akka 2.6.19) has an example:

```scala
def processingFlow(id: Int): Future[Flow[Int, String, NotUsed]] =
  Future {
    Flow[Int].map(n => s"id: $id, value: $n")
  }

val source: Source[String, NotUsed] =
  Source(1 to 10).prefixAndTail(1).flatMapConcat {
    case (List(id: Int), tail) =>
      // base the Future flow creation on the first element
      tail.via(Flow.futureFlow(processingFlow(id)))
  }
```

There are two concerns here.

1. The provided sample flow does not require a `Future`, and therefore serves as a bad example for `futureFlow`. Something async in its place would do the trick.

2. The `case (List(...` fails at runtime (Scala 2.13.8). Replacing `List` with either `Seq` or `Vector` work.

**Comment by raboof** [comment]
>     1. The provided sample flow does not require a `Future`, and therefore serves as a bad example for `futureFlow`. Something async in its place would do the trick.

Yeah - it's often a tricky balance between having a realistic example but also being careful to avoid making the example confusing by adding components that are not essential. Would you like to propose a better example?
 
>     2. The `case (List(...` fails at runtime (Scala 2.13.8). Replacing `List` with either `Seq` or `Vector` work.

Ouch! Would you be interested in providing a PR for that fix?

**Comment by akauppi** [comment]
I'll yield.

While I'm versed with Akka Stream, I don't have a strong feeling on how 1. should be made better ( I ended up not using `.futureFlow` at all). And for 2, I am not aware of repercussions in Scala 2.12, 3.0.
